applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,709 @@
+				err = nrf_modem_gnss_fix_interval_set(interval);
+				if (err) {
+					LOG_ERR("Failed to set fix interval, error: %d", err);
+					return err;
+				}
+				err = nrf_modem_gnss_start();
+				if (err) {
+					LOG_ERR("Failed to start GPS, error: %d", err);
+				} else {
+					ttft_start = k_uptime_get();
+				}
+			} else {
+				err = -EINVAL;
+			}
+		} else if (op == GPS_STOP) {
+			err = nrf_modem_gnss_stop();
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XGPS: (%d,%d),<interval>,<timeout>\r\n",
+			GPS_STOP, GPS_START);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief handle AT#XAGPS commands
+ *  AT#XAGPS=<op>[,<interval>[,<timeout>]]
+ *  AT#XAGPS? READ command not supported
+ *  AT#XAGPS=?
+ */
+int handle_at_agps(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+#if defined(CONFIG_SLM_AGPS)
+	uint16_t op;
+	uint16_t interval;
+	uint32_t timeout = 0;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == AGPS_START) {
+			err = at_params_unsigned_short_get(&at_param_list, 2, &interval);
+			if (err < 0) {
+				return err;
+			}
+			(void)at_params_unsigned_int_get(&at_param_list, 3, &timeout);
+			/* GNSS API spec check */
+			if (interval == 0 || interval == 1 ||
+			   (interval >= 10 && interval <= 65535)) {
+				err = nrf_modem_gnss_fix_interval_set(interval);
+				if (err) {
+					LOG_ERR("Failed to set fix interval, error: %d", err);
+					return err;
+				}
+				/* start with connecting nRF Cloud */
+				err = cloud_connect(nrf_cloud);
+				if (err) {
+					LOG_ERR("Cloud connection failed, error: %d", err);
+				}
+			} else {
+				err = -EINVAL;
+			}
+		} else if (op == AGPS_STOP) {
+			err = cloud_disconnect(nrf_cloud);
+			if (err) {
+				LOG_ERR("Cloud disconnection failed, error: %d", err);
+			}
+			err = nrf_modem_gnss_stop();
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XAGPS: (%d,%d),<interval>,<timeout>\r\n",
+			AGPS_STOP, AGPS_START);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+#endif /* CONFIG_SLM_AGPS */
+
+	return err;
+}
+
+/**@brief handle AT#XPGPS commands
+ *  AT#XPGPS=<op>[,<interval>]
+ *  AT#XPGPS? READ command not supported
+ *  AT#XPGPS=?
+ */
+int handle_at_pgps(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+#if defined(CONFIG_SLM_PGPS)
+	uint16_t op;
+	uint16_t interval;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == PGPS_START) {
+			err = at_params_unsigned_short_get(&at_param_list, 2, &interval);
+			if (err < 0) {
+				return err;
+			}
+			/* P-GPS is used in periodic mode only, in which timeout is ignored */
```

Comment:
Timeout (if you mean fix retry time) is very much valid in periodic mode. The default value of 60 seconds will be used unless the value is configured separately.

Timeout (fix retry) is not valid in continuous navigation mode, but it is valid in single fix and periodic modes.

Reply:
Timeout is NOT fix retry time. It's used in GPS driver only and I forgot to remove them after switching to GNSS APIs. Will do in next commit.

Reply:
The GNSS API has two parameters, "interval" and "fix retry". I think a better name for "fix retry" would be "timeout", but for historical reasons it was not renamed in the new API. Interval tells how often GNSS should try to produce a fix (for example every 30 minutes) and fix retry how long GNSS will keep trying for each fix (for example 60 seconds). If GNSS doesn't get a fix during this time, it will stop trying and will try again when the next fix interval takes place.